### PR TITLE
Hccl rdt

### PR DESCRIPTION
**Pull request by kcmnd**
## Description
This PR adds Ascend NPU support to Ray Direct Transport by using HCCL as the communication backend. It allows Ray to move data directly between Ascend devices with low overhead, similar to how it already works for GPUs. This is needed so Ray can run efficiently on Ascend-based clusters and take advantage of the hardware’s high-speed interconnects for distributed workloads.

## Related issues
Related to: #55381 

## Additional information
Usage example from Ray Direct Transport docs:
```
import torch
import torch_npu
import ray
from ray.experimental.collective import create_collective_group


@ray.remote(resources={"NPU": 1})
class MyActor:
    @ray.method(tensor_transport="hccl")
    def random_tensor(self):
        return torch.randn(1000, 1000).npu()

    def sum(self, tensor: torch.Tensor):
        return torch.sum(tensor)


sender, receiver = MyActor.remote(), MyActor.remote()
group = create_collective_group([sender, receiver], backend="hccl")

# The tensor will be stored by the `sender` actor instead of in Ray's object
# store.
tensor = sender.random_tensor.remote()
result = receiver.sum.remote(tensor)
ray.get(result)
```

### CODE REVIEW

**Review by dayshah — COMMENTED** [review]
hey @kcmnd we're actually going to expose an RDT API that allows users to define their own send/recv transports at runtime and use them in Ray code, similarly to how we allow users to define their own custom serde.

We want to avoid merging a lot of different RDT backends into master that we can't maintain or test in our CI. cc @stephanie-wang @Qiaolin-Yu

**Comment by ji-huazhong** [comment]
> hey @kcmnd we're actually going to expose an RDT API that allows users to define their own send/recv transports at runtime and use them in Ray code, similarly to how we allow users to define their own custom serde.
> 
> We want to avoid merging a lot of different RDT backends into master that we can't maintain or test in our CI. cc @stephanie-wang @Qiaolin-Yu

Hi @dayshah, this RDT API for custom runtime send/recv transports sounds incredibly valuable for our use case—we’ve been looking for a flexible way to integrate custom transport logic into our Ray workflows

Could you share any insights on the release timeline for this feature? We’re eager to plan our integration work and would love to know if there’s a target version or rough timeframe we can align with.

**Comment by kcmnd** [comment]
Hey @dayshah! Any updates on the RDT API?

**Comment by MengjinYan** [comment]
@Sparks0219 Can you help update on the current state of the RDT API? Thanks!
